src/realm/array_mixed.cpp

```diff
@@ -0,0 +1,352 @@
+            m_strings.init_from_ref(ref);
+        }
+        else {
+            m_strings.create();
+            m_strings.update_parent();
+        }
+    }
+}
+
+void ArrayMixed::replace_index(size_t old_ndx, size_t new_ndx, size_t payload_arr_index)
+{
+    if (old_ndx != new_ndx) {
+        size_t sz = m_composite.size();
+        for (size_t i = 0; i != sz; i++) {
+            int64_t val = m_composite.get(i);
+            if (size_t((val & s_payload_idx_mask) >> s_payload_idx_shift) == payload_arr_index &&
+                (val >> s_data_shift) == int64_t(old_ndx)) {
+                m_composite.set(i, int64_t(new_ndx << s_data_shift) | (val & 0xff));
+                return;
+            }
+        }
+    }
+}
+
+void ArrayMixed::erase_linked_payload(size_t ndx)
+{
+    auto val = m_composite.get(ndx);
+    auto payload_arr_index = size_t((val & s_payload_idx_mask) >> s_payload_idx_shift);
+
+    if (payload_arr_index) {
+        // A value is store in one of the payload arrays
+        size_t last_ndx = 0;
+        size_t erase_ndx = size_t(val >> s_data_shift);
+        // Clean up current value by moving last over
+        switch (DataType((val & s_data_type_mask) - 1)) {
+            case type_Int:
+            case type_Float:
+            case type_Double: {
+                ensure_int_array();
+                last_ndx = m_ints.size() - 1;
+                if (erase_ndx != last_ndx) {
+                    m_ints.set(erase_ndx, m_ints.get(last_ndx));
+                    replace_index(last_ndx, erase_ndx, payload_arr_index);
+                }
+                m_ints.erase(last_ndx);
+                break;
+            }
+            case type_String:
+            case type_Binary: {
+                ensure_string_array();
+                last_ndx = m_strings.size() - 1;
+                if (erase_ndx != last_ndx) {
+                    StringData tmp = m_strings.get(last_ndx);
+                    std::string tmp_val(tmp.data(), tmp.size());
+                    m_strings.set(erase_ndx, StringData(tmp_val));
+                    replace_index(last_ndx, erase_ndx, payload_arr_index);
+                }
+                m_strings.erase(last_ndx);
+                break;
+            }
+            case type_Timestamp: {
+                ensure_int_pair_array();
+                last_ndx = m_int_pairs.size() - 2;
+                erase_ndx <<= 1;
+                if (erase_ndx != last_ndx) {
+                    m_int_pairs.set(erase_ndx, m_int_pairs.get(last_ndx));
+                    m_int_pairs.set(erase_ndx + 1, m_int_pairs.get(last_ndx + 1));
+                    replace_index(last_ndx >> 1, erase_ndx >> 1, payload_arr_index);
+                }
+                m_int_pairs.truncate(last_ndx);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+}
+
+int64_t ArrayMixed::store(const Mixed& value)
+{
+    DataType type = value.get_type();
+    int64_t val;
+    switch (type) {
```

Comment:
Get/set are missing handling `type_Link` for storing `ObjKey`

Reply:
I suggest adding cases for all types explicitly here and in `get` and having no default case. The unsupported types could use `REALM_UNREACHABLE()`.

Reply:
I was not planing to support that just yet.